src/modules/account/store/reducer.js

```diff
@@ -58,14 +58,33 @@ export const list = (state = {}, { type, encryptedAccount, accountDetail, addres
   }
 };
 
+export const localNonce = (state = {}, { type, address, nonce, transactionHex }) => {
+  switch (type) {
+    case actionTypes.setAccountNonce:
+      if (state[address]?.[transactionHex]) {
+        return state;
+      }
+      return {
+        ...state,
+        [address]: {
+          ...state[address],
+          [transactionHex]: nonce,
+        },
+      };
```

Comment:
As a follow up to my above statement this create a history of transactions created, though there is not clean up strategy.